dashboard/src/modules/components/OverviewComponent/NewRunsComponent.jsx

```diff
@@ -37,26 +37,31 @@ const NewRunsComponent = () => {
   const { newRuns, initNewRuns, selectedRuns } = useSelector(
     (state) => state.overview
   );
-  const loginDetails = useSelector((state) => state.userAuth.loginDetails);
+  // const loginDetails = useSelector((state) => state.userAuth.loginDetails);
 
   const [perPage, setPerPage] = useState(ROWS_PER_PAGE);
   const [page, setPage] = useState(START_PAGE_NUMBER);
 
-  const onSetPage = (_evt, newPage, _perPage, startIdx, endIdx) => {
-    setPage(newPage);
-    dispatch(setRows(newRuns.slice(startIdx, endIdx)));
-  };
-
+  const onSetPage = useCallback(
+    (_evt, newPage, _perPage, startIdx, endIdx) => {
+      setPage(newPage);
+      dispatch(setRows(newRuns.slice(startIdx, endIdx)));
+    },
+    [dispatch, newRuns]
+  );
```

Comment:
I need some help understanding the implications of this change.

It looks to me like all that happened is that we took a function (`onSetPage()`) and put it inside a closure which captures the values of `dispatch` and `newRuns`...what did that do for us?

I _think_ that `onSetPage()` already had access to the _current_ value of `newRuns` (I'm not concerned about the value of `dispatch` changing), so, by using the closure, I'm thinking that you've ensured that the value of `newRuns` which the function sees will **not** change when `newRuns` is updated...which seems...dangerous.  Is this what is supposed to happen and what you wanted?

Ditto for `onPerPageSelect()`.